### PR TITLE
Results of AttributeConverters should be mutable to allow dirty checking in H6

### LINKED ISSUE #16551: Quarkus 3: RealmModelTest.testRealmLocalizationTexts fails
BLOCKED: see https://github.com/keycloak/keycloak/issues/16551#issuecomment-1400633351

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

testsuite

### Describe the bug

The `org.keycloak.testsuite.model.RealmModelTest.testRealmLocalizationTexts` test fails in the quarkus3 branch in `legacy-jpa` profile

### Version

quarkus3

### Expected behavior

Test passes

### Actual behavior

Test fails

### How to Reproduce?

Run the test in `legacy-jpa` profile

### Anything else?

https://github.com/keycloak/keycloak/actions/runs/3966487788/jobs/6799050521

### PR DESCRIPTION
**Pull request by ahus1**
Relates to #16551

### CODE REVIEW

**Comment by ahus1** [comment]
@hmlnarik - from what it looks today, this change can be either applied to the Q3 branch, or can be integrated in to main. 

**TL;DR:** I would keep it against the Q3 branch for now. Please let me know if you see this differently.

The longer version: Until further discussions with the Hibernate team, I see no technical advantage when running in main with the change. The only advantage I can see is that we wouldn't need to rebase it later ... at the same time there is no way to test if this works in the main branch, so we need this PR to show that the tests are green.

**Comment by hmlnarik** [comment]
I agree with your reasoning, let's keep it in `quarkus3` branch for now.

**Review by vramik — APPROVED** [review]
Tested locally, thank you @ahus1!

**Review by hmlnarik — APPROVED** [review]
Thanks for the solution!
